### DISCUSSION TITLE
Controlling Precision to Avoid Dropping Thin Mesh Components

### DISCUSSION THREAD

**Discussion by Eric-Vin** [elalish/manifold#649] [Q&A] [answered]
It seems like very thin pieces of my mesh are being dropped by Manifold. Is there a way to control the precision so this doesn't happen (ideally in the Python bindings)? I can provide an example if that would be helpful.

**Comment by elalish** [comment]
Yes, an example would be helpful. Is it related to #560? In general this is a feature of manifold (removing thin cruft from inexact operations). Since our library operates on volumes, thin surfaces don't have much meaning.

**Comment by Eric-Vin** [comment]
Thanks for the response! I'm not sure if it's related to the issue, but here's an example. I'm using Trimesh here but all the intersection operation does in this case is call directly out to Manifold. I want to construct regions representing visible regions for objects. As part of the testing I was generating very thin regions, and I noticed that part of the regions were being dropped. Here's what I am getting, with the origin not contained in the mesh, versus what I should be getting, with the origin contained in the mesh.

![image](https://github.com/elalish/manifold/assets/8935814/31eaca3b-580c-461d-8ee3-c947271c445f)
![image](https://github.com/elalish/manifold/assets/8935814/43b544dd-2069-4615-b753-ee69d0ac0cb8)

Here's the code and meshes to replicate this:
```
import trimesh

vs = trimesh.load("ViewSection.obj")
bs = trimesh.load("Sphere.obj")

vr = vs.intersection(bs)
vr.show()
```
[Meshes.zip](https://github.com/elalish/manifold/files/13690330/Meshes.zip)

The key thing here for me is that there are points in `vs` and `bs` (like (0,0,0)), that `vr` does not contain. Any advice?

**Reply by pca006132** [reply, marked as answer]
OK, I have a look at this, this is due to how precision is handled. The view section you have has a large x coordinate, which makes the precision large (0.11459, basically 1e-6*max abs coordinate), so anything smaller than this will be removed by manifold.
There are two solutions for this. The first one would be to increase the precision, i.e. with #542. The other method which works now is to crop the view section to the bounding box of the sphere, and manually override the precision by reimporting the mesh (because cropping is intersection, and this will not improve precision in general).

Example code that works with the second method:
```c++
  // somehow I cannot get obj import to work
  Manifold m1 = ImportMesh("./Sphere.glb");
  Manifold m2 = ImportMesh("./ViewSection.glb");
  Manifold m1BoundingBox = Manifold::Cube(m1.BoundingBox().Size(), true).Translate(m1.BoundingBox().Center());
  // note that m2 may be "optimized" by manifold in this operation... it does not happen in this case though
  m2 ^= m1BoundingBox;
  MeshGL m2Mesh = m2.GetMeshGL();
  // clear the precision, so it is recomputed from vertex coordinates (which are much smaller now)
  m2Mesh.precision = -1;
  m2 = Manifold(m2Mesh);

  Manifold m3 = m1 ^ m2;
  printf("precision = %.5f\n", m3.Precision());
  std::cout << m3.BoundingBox() << std::endl;
  ExportMesh("intersection.obj", m3.GetMeshGL(), {});
```

**Reply by Eric-Vin** [reply]
This makes sense, thanks so much for the response! I'll try resolving this using your method and/or change how I'm constructing the meshes to avoid this.
